delete mydumper doc in dev and 5.0

lichunzhu opened the pull request:
<!--Thanks for your contribution to TiDB documentation. Please answer the following questions.-->

### What is changed, added or deleted? (Required)

<!--Tell us what you did and why.-->
 Remove mydumper-related content from the master branch as much as possible.

+ Delete `backup-and-restore-using-mydumper-lightning.md`
+ Delete `mydumper-overview.md`
+ Replace `mydumper` with `Dumpling` in some docs.

### Which TiDB version(s) do your changes apply to? (Required)

<!-- You **must** choose the TiDB version(s) that your changes apply to. Fill in "x" in [] to tick the checkbox below.-->

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions) **version-specific changes: remove aliases.**
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

<!-- For contributors with **WRITE ACCESS** in this repo:
If you select **two or more** versions from above, to trigger the bot to cherry-pick this PR to your desired release branch(es), you **must** add labels such as "needs-cherry-pick-4.0", "needs-cherry-pick-3.1", "needs-cherry-pick-3.0", or "needs-cherry-pick-2.1" on the right side of this PR page.-->

### What is the related PR or file link(s)?

<!--Give us some reference link(s) that might help quickly review and merge your PR.-->

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

<!-- Provide as much information as possible so that reviewers can review your changes more efficiently.
If you are not sure of the options, leave it as it is. -->

- [x] Delete files
- [x] Change aliases
- [x] Have version specific changes <!-- If yes, please add the label "version-specific-changes-required"-->
- [x] Might cause conflicts

nullnotnil commented:
Can you add an alias in dumpling's metadata so that it will redirect bookmarks / SEO? Otherwise LGTM

nullnotnil commented:
CI is also reporting the following pages link here:
```
=== ERROR REPORT == :

FILE: ./get-started-with-tidb-lightning.md
[✖] /home/circleci/pingcap/docs-cn//mydumper-overview.md

6 links checked.
[✖] /home/circleci/pingcap/docs-cn//mydumper-overview.md → Status: 400
FILE: ./migrate-from-mysql-mydumper-files.md
[✖] /home/circleci/pingcap/docs-cn//mydumper-overview.md

6 links checked.
[✖] /home/circleci/pingcap/docs-cn//mydumper-overview.md → Status: 400
FILE: ./faq/migration-tidb-faq.md
[✖] /home/circleci/pingcap/docs-cn//mydumper-overview.md

10 links checked.
[✖] /home/circleci/pingcap/docs-cn//mydumper-overview.md → Status: 400
FILE: ./tidb-lightning/tidb-lightning-faq.md
[✖] /home/circleci/pingcap/docs-cn//mydumper-overview.md

13 links checked.
[✖] /home/circleci/pingcap/docs-cn//mydumper-overview.md → Status: 400
FILE: ./tidb-lightning/deploy-tidb-lightning.md
[✖] /home/circleci/pingcap/docs-cn//mydumper-overview.md

12 links checked.
[✖] /home/circleci/pingcap/docs-cn//mydumper-overview.md → Status: 400
FILE: ./backup-and-restore-using-mydumper-lightning.md
[✖] /home/circleci/pingcap/docs-cn//mydumper-overview.md

6 links checked.
[✖] /home/circleci/pingcap/docs-cn//mydumper-overview.md → Status: 400
FILE: ./download-ecosystem-tools.md
[✖] /home/circleci/pingcap/docs-cn//mydumper-overview.md

13 links checked.
[✖] /home/circleci/pingcap/docs-cn//mydumper-overview.md → Status: 400
```

lichunzhu commented:
> Can you add an alias in dumpling's metadata so that it will redirect bookmarks / SEO? Otherwise LGTM

It looks like mydumper doc has many references. I will try to fix https://github.com/pingcap/docs-cn/pull/4857 at first and then merge this PR.

TomShawn commented:
@lichunzhu Does it mean mudumper is deprecated (can no longer be used) in dev?

nullnotnil commented:
> @lichunzhu Does it mean mydumper is deprecated (can no longer be used) in dev? If so, in addition to deleting all references, we should also add an alias to the document which we want the mydumper users to read (I guess it's Dumpling) and tell them that mydumper is superseded by Dumpling in the lastest dev version.

For consistent terminology usage, I would say it is deprecated in stable (for which there is already a warning) and removed in dev.

TomShawn commented:
> > @lichunzhu Does it mean mydumper is deprecated (can no longer be used) in dev? If so, in addition to deleting all references, we should also add an alias to the document which we want the mydumper users to read (I guess it's Dumpling) and tell them that mydumper is superseded by Dumpling in the lastest dev version.
> 
> For consistent terminology usage, I would say it is deprecated in stable (for which there is already a warning) and removed in dev.

`Deprecated` means that users have no way of using it because we don't let them. In fact, we don't go that far and they can still use it. So we only say "strongly recommended" to use or not to use, or "no longer provide any support" for the old method.

Also, if `mydumper-overview.md` is to be deleted, we should migrate the warning to `dumpling-overview.md`.

nullnotnil commented:
> `Deprecated` means that users have no way of using it because we don't let them. In fact, we don't go that far and they can still use it. So we only say "strongly recommended" to use or not to use, or "no longer provide any support" for the old method.

[Wikipedia agrees](https://en.wikipedia.org/wiki/Deprecation) with me here:

> In several fields, deprecation is the discouragement of use of some terminology, feature, design, or practice, typically because it has been superseded or is no longer considered efficient or safe, without completely removing it or prohibiting its use. Typically, deprecated materials are not completely removed to ensure legacy compatibility or back up practice in case new methods are not functional in an odd scenario.
> It can also imply that a feature, design, or practice will be removed or discontinued entirely in the future.

kennytm commented:
@TomShawn  I still prefer the classification in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Deprecated_and_obsolete_features. "Deprecated" is fine IMO.

> … deprecated (that is, still available but planned for removal) and obsolete (that is, no longer usable).

TomShawn commented:
@nullnotnil @kennytm Thanks for explaining! Let's use `deprecated`.

TomShawn commented on migrate-from-mysql-mydumper-files.md:
```diff
@@ -13,7 +13,7 @@ aliases: ['/docs-cn/dev/migrate-from-mysql-mydumper-files/']
 
 > **注意：**
-> 
+>
```
```suggestion
> 
```

TomShawn replied:
```suggestion
> 
```

lichunzhu commented:
@TomShawn @kennytm PTAL

kennytm reviewed (commented):
LGTM

TomShawn commented on backup-and-restore-using-dumpling-lightning.md:
```diff
@@ -1,10 +1,14 @@
 ---
 title: 使用 Dumpling/TiDB Lightning 进行备份与恢复
-aliases: ['/docs-cn/dev/backup-and-restore-using-dumpling-lightning/','/docs-cn/dev/export-or-backup-using-dumpling/','/zh/tidb/dev/export-or-backup-using-dumpling']
+aliases: ['/docs-cn/dev/backup-and-restore-using-dumpling-lightning/','/docs-cn/dev/export-or-backup-using-dumpling/','/zh/tidb/dev/export-or-backup-using-dumpling','/docs-cn/dev/backup-and-restore-using-mydumper-lightning/','/docs-cn/dev/how-to/maintain/backup-and-restore/mydumper-lightning/','/docs-cn/dev/how-to/maintain/backup-and-restore/']
```
```suggestion
aliases: ['/docs-cn/dev/backup-and-restore-using-dumpling-lightning/','/docs-cn/dev/export-or-backup-using-dumpling/','/zh/tidb/dev/export-or-backup-using-dumpling','/zh/tidb/dev/backup-and-restore-using-mydumper-lightning']
```

TomShawn replied:
```suggestion
aliases: ['/docs-cn/dev/backup-and-restore-using-dumpling-lightning/','/docs-cn/dev/export-or-backup-using-dumpling/','/zh/tidb/dev/export-or-backup-using-dumpling','/docs-cn/dev/backup-and-restore-using-mydumper-lightning/','/docs-cn/dev/how-to/maintain/backup-and-restore/mydumper-lightning/','/docs-cn/dev/how-to/maintain/backup-and-restore/']
```

TomShawn commented on dumpling-overview.md:
```diff
@@ -1,7 +1,7 @@
 title: Dumpling 使用文档
 summary: 使用 Dumpling 从 TiDB 导出数据。
-aliases: ['/docs-cn/dev/dumpling-overview/']
+aliases: ['/docs-cn/dev/dumpling-overview/','/zh/tidb/dev/mydumper-overview/']
```
```suggestion
aliases: ['/docs-cn/dev/dumpling-overview/','/docs-cn/dev/mydumper-overview/','/docs-cn/dev/reference/tools/mydumper/','/zh/tidb/dev/mydumper-overview/']
```

TomShawn commented:
@lichunzhu Should we also delete loader-related content in dev?

overvenus commented on download-ecosystem-tools.md:
```diff
@@ -70,7 +70,7 @@ aliases: ['/docs-cn/dev/download-ecosystem-tools/','/docs-cn/dev/reference/tools
 ## Syncer，Loader 和 Mydumper
 
-如需下载最新版本的 [Syncer](/syncer-overview.md)，[Loader](/loader-overview.md) 或 [Mydumper](/mydumper-overview.md)，直接下载 tidb-enterprise-tools 安装包即可，因为这些工具均包含在此安装包中。
+如需下载最新版本的 [Syncer](/syncer-overview.md)，[Loader](/loader-overview.md) 或 Mydumper，直接下载 tidb-enterprise-tools 安装包即可，因为这些工具均包含在此安装包中。
```
```suggestion
如需下载最新版本的 [Syncer](/syncer-overview.md)，[Loader](/loader-overview.md) 或 Dumpling，直接下载 tidb-enterprise-tools 安装包即可，因为这些工具均包含在此安装包中。
```

Still "Mydumper"?

lichunzhu replied:
It's still `Mydumper` here. Dumpling is not in the `tidb-enterprise-tools` package

overvenus reviewed (commented):
Rest LGTM

TomShawn commented on download-ecosystem-tools.md:
```diff
@@ -70,7 +70,7 @@ aliases: ['/docs-cn/dev/download-ecosystem-tools/','/docs-cn/dev/reference/tools
 ## Syncer，Loader 和 Mydumper
 
-如需下载最新版本的 [Syncer](/syncer-overview.md)，[Loader](/loader-overview.md) 或 [Mydumper](/mydumper-overview.md)，直接下载 tidb-enterprise-tools 安装包即可，因为这些工具均包含在此安装包中。
+如需下载最新版本的 [Syncer](/syncer-overview.md)，[Loader](/loader-overview.md) 或 Mydumper，直接下载 tidb-enterprise-tools 安装包即可，因为这些工具均包含在此安装包中。
```
This can be a little confusing because the `tidb-enterprise-tools` package still contains mydumper. Should we add a note below to discourage users from using mydumper/loader/Syncer and provide the recommended alternatives?

lichunzhu replied:
We have warnings in `Syncer` and `Loader` documents. I think we needn't add them here.

lichunzhu commented:
> @lichunzhu Should we also delete loader-related content in dev?

loader is also deprecated and replaced by `tidb-lightning`. So I think we don't need to change that.

TomShawn commented:
> > @lichunzhu Should we also delete loader-related content in dev?
> 
> loader is also deprecated and replaced by `tidb-lightning`. So I think we don't need to change that.

mydumper is also deprecated and replaced by Dumpling. Why do we remove mydumper while keeping loader? And also the deprecated Syncer.

lichunzhu commented:
> > > @lichunzhu Should we also delete loader-related content in dev?
> > 
> > 
> > loader is also deprecated and replaced by `tidb-lightning`. So I think we don't need to change that.
> 
> mydumper is also deprecated and replaced by Dumpling. Why do we remove mydumper while keeping loader? And also the deprecated Syncer.

Then maybe we can also delete `loader` and `syncer` in dev branch. But I think we can do this in another PR.

TomShawn commented:
According to the PM @scsldb's discretion, it is not a good idea to remove all mydumper-related content from the `dev` version because it is an issue of lifecycle management, which is accepted by @lichunzhu and @kissmydb. So I'll close this PR for now.

TomShawn commented on dumpling-overview.md:
```diff
@@ -10,6 +10,10 @@ aliases: ['/docs-cn/dev/dumpling-overview/']
 
+> **注意：**
+>
+> PingCAP 之前维护的 Mydumper 工具 fork 自 [mydumper project](https://github.com/maxbube/mydumper)，针对 TiDB 的特性进行了优化。Mydumper 目前已经不再开发新功能，其绝大部分功能已经被 [Dumpling](/dumpling-overview.md) 取代。Dumpling 工具使用 Go 语言编写，支持更多针对 TiDB 特性的优化。请切换到 Dumpling。
```
```suggestion
> PingCAP 之前维护的 Mydumper 工具 fork 自 [mydumper project](https://github.com/maxbube/mydumper)，针对 TiDB 的特性进行了优化。Mydumper 目前已经不再开发新功能，其绝大部分功能已经被 [Dumpling](/dumpling-overview.md) 取代。Dumpling 工具使用 Go 语言编写，支持更多针对 TiDB 特性的优化。请切换到 Dumpling。
>
> 如果你需要阅读 Mydumper 的文档，可参阅 [Mydumper 使用文档](https://docs.pingcap.com/zh/tidb/v4.0/mydumper-overview)。
```

TomShawn commented on backup-and-restore-using-dumpling-lightning.md:
```diff
@@ -1,10 +1,14 @@
 
+> **注意：**
+>
+> PingCAP 之前维护的 Mydumper 工具 fork 自 [mydumper project](https://github.com/maxbube/mydumper)，针对 TiDB 的特性进行了优化。Mydumper 目前已经不再开发新功能，其绝大部分功能已经被 [Dumpling](/dumpling-overview.md) 取代。Dumpling 工具使用 Go 语言编写，支持更多针对 TiDB 特性的优化。请切换到 Dumpling。
```
```suggestion
> PingCAP 之前维护的 Mydumper 工具 fork 自 [mydumper project](https://github.com/maxbube/mydumper)，针对 TiDB 的特性进行了优化。Mydumper 目前已经不再开发新功能，其绝大部分功能已经被 [Dumpling](/dumpling-overview.md) 取代。Dumpling 工具使用 Go 语言编写，支持更多针对 TiDB 特性的优化。请切换到 Dumpling。
>
> 如果你需要阅读使用 Mydumper/TiDB Lightning 进行备份与恢复的文档，可参阅[使用 Mydumper/TiDB Lightning 进行备份与恢复](https://docs.pingcap.com/zh/tidb/v4.0/backup-and-restore-using-mydumper-lightning)。
```

yikeke commented on backup-and-restore-using-dumpling-lightning.md:
```diff
@@ -1,10 +1,16 @@
 ---
 title: 使用 Dumpling/TiDB Lightning 进行备份与恢复
-aliases: ['/docs-cn/dev/backup-and-restore-using-dumpling-lightning/','/docs-cn/dev/export-or-backup-using-dumpling/','/zh/tidb/dev/export-or-backup-using-dumpling']
+aliases: ['/docs-cn/dev/backup-and-restore-using-dumpling-lightning/','/docs-cn/dev/export-or-backup-using-dumpling/','/zh/tidb/dev/export-or-backup-using-dumpling','/docs-cn/dev/backup-and-restore-using-mydumper-lightning/','/docs-cn/dev/how-to/maintain/backup-and-restore/mydumper-lightning/','/docs-cn/dev/how-to/maintain/backup-and-restore/','/backup-and-restore-using-mydumper-lightning/']
```
'/backup-and-restore-using-mydumper-lightning/' -> '/zh/tidb/dev/backup-and-restore-using-mydumper-lightning/'

TomShawn replied:
Done in https://github.com/pingcap/docs-cn/pull/5208.
